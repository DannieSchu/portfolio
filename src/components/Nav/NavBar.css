.NavBar {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  
  position: fixed;
  height: 100vh;
  width: 6rem;
  z-index: 1;
  
  background-color: var(--charcoal);
}
