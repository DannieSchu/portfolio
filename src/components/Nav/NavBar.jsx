import React from 'react';
import NavLink from './NavLink';
import { navIconsData } from '../../data/navIconsData';
import styles from './NavBar.css';

const NavBar = () => {
  const navElements = navIconsData.map((element, i) => <NavLink key={i} {...element} />);

  return (
    <nav className={styles.NavBar}>
      {navElements}
    </nav>
  );
};

export default NavBar;
