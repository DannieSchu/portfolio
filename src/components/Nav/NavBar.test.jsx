import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar';

describe('NavBar component', () => {
  it('renders NavBar', () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
