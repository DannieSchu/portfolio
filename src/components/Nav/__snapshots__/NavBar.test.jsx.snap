// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders NavBar 1`] = `
<nav
  className="NavBar"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <NavLink
    activeIcon="/path/image"
    icon="/path/image"
    key="0"
    to="Home"
  />
  <NavLink
    activeIcon="/path/image"
    icon="/path/image"
    key="1"
    to="About"
  />
  <NavLink
    activeIcon="/path/image"
    icon="/path/image"
    key="2"
    to="Projects"
  />
  <NavLink
    activeIcon="/path/image"
    icon="/path/image"
    key="3"
    to="Contact"
  />
</nav>
`;
